src/arch/fake_time/CMakeLists.txt

```diff
@@ -14,8 +14,13 @@ if(FORTE_FAKE_TIME)
 
   forte_add_definition("-DFORTE_FAKE_TIME")
```

Comment:
This definition is used to conditionally compile exchanged `getNanoSecondsMonotonic()` and `getNanoSecondsRealtime()` functions via the architecture dependent `forte_architecture_time.h` whenever FAKE_TIME is enabled during compilation;

I am not sure that/how this is handled by this change?

Reply:
I managed to find a solution. 

There are three functions that were "replaced" by the fake timer, namely: forte_timer, getNanoSecondsMonotonic and getNanoSecondsRealtime. Now, we have an `Arch` version of them in each architecture (forte_timer_arch, getNanoSecondsMonotonicArch and getNanoSecondsRealtime), which is the same as the original. These functions are always compiled. 

The fake timer offers now three parallel functions: forte_timer_fake, getNanoSecondsMonotonicFake and getNanoSecondsRealtimeFake, which are compiled, as before, only if the FAKE_TIMER is enabled. These fake timer functions have now a conditional which checks which one is the currently timer being used (fake timer or the "regular" one form the archictures) and depending on that it calls the `Arch` version of these functions or the original fake implementation.

The `forte_architecture_time.h` offers alias to the original functions (without `Arch` nor `fake`) which are set to the `Arch` or the `Fake` version of them depending if the FAKE_TIMER is enabled. 

Performance-wise, if the FAKE_TIMER is disabled, nothing changes. If it's enabled, there's an extra `if`.

Reply:
Yes, that looks good - I'm still not fully happy with the string comparison in that extra `if` - I see that this makes the code a lot more generic but it feels like a rather high price to pay as I don't see how this could be removed by optimization

as of now there are only two possible timers (`architecture` and "faketimer`) for a compiled binary - maybe an enum would be the extensible choice? or is this still overthinking the problem at hand?

Reply:
I like the `enum` idea. This will make the comparison faster. Regardgin the extra `if`, it comes only if you compile with `fake_time` which you usually do while testing, you don't deploy on real scenarios. Also, before this PR, if you compiled `fake_time` you didn't have any other timer, so now you have both at the price of an extra `if` when testing basically.